enhancements/sig-architecture/19-projected-serviceaccount-token/README.md

```diff
@@ -0,0 +1,159 @@
+
+
+## Goals & Non-Goals
+
+### Goals
+
+- Manage the lifecycle of service-accounts across multiple cluster.
+- Persist the token of the service-account back to the hub cluster.
+
+### Non-Goals
+
+- The addon will not be an external token signer.
+- The addon will not be managing rbac authorization for the managed tokens.
+
+## Design
+
+### Component & API
+
+We purpose to add two components by this design:
+
+1. An addon manager based on addon-framework in the hub cluster.
+2. Addon agents in each of the managed cluster.
+
+and additionally there will also be a few new custom resources:
+
+A sample of the `ManagedServiceAccount` will be:
+
+```yaml
+# `ManagedServiceAccount`is cluster-scoped
+apiVersion: proxy.open-cluster-management.io/v1alpha1
+kind: ManagedServiceAccount
+metadata:
+  name: my-foo-operator-serviceaccount
+spec:
+  provision: 
+    namespace: default
+    name: my-foo-operator-serviceaccount
+  audience: operator
+  rotation:
+    interval: 3000s
+    tokenMaxValidTime: 3600s
+status:
+  conditions:
+    - type: AllServiceAccountPresent
+      status: True
+```
+
+A sample of the reported/persisted service-account token will be:
+
+(We make the token presisted as a secret resource here so that users
+can mount it easier)
+
+```yaml
+apiVersion: v1
+kind: Secret
+metadata: 
+  ownerReferences:
+    - apiVersion: proxy.open-cluster-management.io/v1alpha1
+      resource: managedserviceaccounts
+      name: my-foo-operator-serviceaccount
+  name: my-foo-operator-serviceaccount
+  namespace: cluster-1
+data:
+  ca.crt: ...
+  token: ...
+```
+
+Based on the extensibility provided by the addon-framework, the addon manager
+will be doing the following things upon detecting the presence of the addon:
+
+- Install addon-agent to each managed cluster.
+- Grant the addon-agent the permission of writing secrets to a highly-secured
+  namespace in the hub cluster.
+- Grant the addon-agent the permission of list-watching `ManagedServiceAccount`
+  in the hub cluster.
+- Grant addon-agent the permission of writing service-account in the managed
+  cluster.
+- Grant addon-agent the permission of issuing `authenticationv1.TokenRequest`
+  in the managed cluster.
+  
+Then addon-agent will be list-watch after started and dynamically provisions
+local service-accounts and persist the token to the hub cluster based on the 
+rotation policy.
+
+### A Highly-secured namespace for persisting the tokens
+
+The addon-agents need to find a place to persist the tokens after they finally
+succeed obtaining tokens via `authenticationv1.TokenRequest` api. In this 
+proposal, we're expecting all these tokens to be persisted in a single 
+highly-secured namespace such as `managed-cluster-credentials`. To protect the 
+tokens from being copied for other unexpected usage, the namespace should not
+be accessible to most of the hub cluster users except:
+
+1. the addon-agent for rotation the token. (write-only to its managing secret)
```

Comment:
can you be specific about what verbs?  YOu cannot have named create, so you're thinking about a named apply?

Reply:
oh rbac doesnt support named create, is it? so yes, a named server-side-apply can be better than named create.